cmd/uplink/cmd/cp.go

```diff
@@ -88,6 +90,17 @@ func upload(ctx context.Context, bs buckets.Store, srcFile string, destObj *url.
 		return err
 	}
 
+	/* create a signal of type os.Signal */
+	c := make(chan os.Signal, 0x01)
+
+	/* register for the os signals */
+	signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
+
+	go func() {
+		<-ctx.Done()
+		utils.LogClose(f)
```

Comment:
Isn't this redundant? We already have `defer utils.LogClose(f)` on line 70.

Reply:
my thought process of clean handing of CTRL+C. 
1. close the stream as FAST as you can, that way the downstream( streams, segments and ec) has no more processing to be done. 
2. Without the above code, I saw from the CTRL+C and actually handling/passing of cancel() to downstream, there was considerable delay(based on what CPU was doing at that instance), and during this delay, some segments were still erasure encoded and uploaded. .. I tried with playing a download dummy video and a ran bunch of applications (started up), tried as much to increase CPU utilization and hit the CTRL+C.  I saw a delay and in that delay it uploaded few segments. But I wanted to stop that, so I added the code in line 101 and saw a considerable improvement. 

I was expecting to see "file already closed error",  but I haven't seen that error. my thought process closing the stream saves  some of the CPU processing, upload bandwidth at client, satellite and piece node's storage space and its bandwidth. Might not be much when looked at one client. If million people using, then the numbers become significant. That was my rational and thought it was not redundant.  I will remove it.